### DISCUSSION TITLE
Big Ticket Items

### DISCUSSION THREAD

**Discussion by vladmandic** [vladmandic/sdnext#1195] [Polls]
From two big ticket items on my plate, which one would you like to see more?  
- Enabling usage of tensor cores on nVidia GPUs for ~2x performance increase  
  For example, **TensorRT** or any other newer methods like AITemplate, etc.  
  Note this is applicable to new nVidia GPUs only (3xxx and above)  
- Enabling usage of new models other than standard StableDiffusion 1.5 /2.1
  For example, **DeepFloyd**, **Kandinsky**, etc.  
  Note this is applicable to all GPUs  

Note that in both cases, existing workflow will be limited.
For example
- LORA/LYCO are very Torch and SD specific thing and are not transferrable
- Advanced model control such as ControlNet is also not transferrable easily

Eventually, I hope all is going to get done, but need to prioritize...

**Comment by Aptronymist** [comment]
While I am now an nVidia user, I was sold on this project when I saw that Vlad fully intended to integrate AMD support (then I used rx590, still secondary) as best he could, which he did, I think he's done the same for Intel's ARC and M1s as best as he could as well. He didn't just ignore it and leave that to be handled by a fork.
 
I feel it would be a bit too... self-interested to focus on one particular GPU manufacturer, even if (or because) that is what the majority are using. 
I'd love a performance boost, who wouldn't, but there's a slew (possibly multiple slews) of emerging models that open new doors to what we all seek to do with this technology. I personally feel it would be selfish of me to vote for improving my own performance when we could all have and enjoy things like DeepFloyd and Kandinsky for everyone under the all-encompassing Vlad-brella™️ of SD.Next.

**Reply by Lumpsack** [reply]
Thank you - personally I've a vested interest - being an AMD user (stupid me didn't find out about the limitations until after buying, going for the tonne of VRAM for the money, hey hum... wrong call), so very much appreciate your choice.

**Reply by Aptronymist** [reply]
I want my AMD brothers right beside me making art! There shouldn't be any second-class citizens when it comes to AI artwork.

**Reply by iDeNoh** [reply]
I whole heartily agree with you, while faster gens on the 12 or so cards that would benefit from trt is technically cool, I think supporting other models would be MUCH better for this fork as a whole, it's not like the idea is you get one or the other either, it's just which comes first. I think raw speed is generally less useful for most people, but I don't think a majority of users see it that way, which kinda sucks.

**Reply by AmmBe** [reply]
Adding new models seems to me the most difficult task. Without fine tuning, they are all inferior to stable diffusion 1.5. This means that they will still need to add a way to train them, let me remind you that even the version of stable diffusion 2.1 is still not used by anyone. Also optimize, for example DeepFloyd requires 16gb VRAM. Still it will be necessary to maintain all this hodgepodge in working order.
But the possibility of using tensor cores in the workflow is more logical, including for future models.

**Reply by iDeNoh** [reply]
The limitations for using alternative models is the same if not worse than supporting tensor cores, only a handful of devices support them and they can't use any of the fine tuning techniques we use in 1.5, I get wanting moar speed, but I wholeheartedly disagree that it's more logical than starting the groundwork for supporting alternative models that everyone can benefit from.

**Reply by vladmandic** [reply]
> But the possibility of using tensor cores in the workflow is more logical, including for future models.

no, any tensor cores work would be very specific to existing sd 1.5/2.1 implementation. there is zero chance that would be applicable to other models.

**Comment by xedis** [comment]
Will backwards compatibility be available in both cases?
- example: Would LoRA/LyCo still be usable when using SD models OR if tensor cores were not enabled?

**Reply by vladmandic** [reply]
of course!

**Comment by drax-xard** [comment]
While I'd like to have the speed of TensorRT, I think that implementing alternative model architectures will help draw attention to them and hopefully bring improvements and optimizations like those that SD gained from the community's work.

**Comment by shutter-cat** [comment]
I would LOVE Kandinsky model support, it provide really interesting images, also 2.2 version coming soon...

**Comment by TotalDay** [comment]
Intel ARC have huge tensor cores
Can your make support fot Intel ARc tensor cores support?

**Reply by vladmandic** [reply]
that would be up to intel ipex engine.

**Comment by TotalDay** [comment]
Intel ARC 750 - 448 tensor cores
Intel ARC 770 - 512 tensor cores
NVIDIA GeForce RTX 4070 - 184 Tensor Cores
NVIDIA GeForce RTX 4080 - 304 Tensor Cores
NVIDIA GeForce RTX 4090 - 512 Tensor Cores

Please add tensor cores support for intel ARC!!!

**Reply by Disty0** [reply]
You can't compare Tensor cores from different architectures.
A770's theoretical maximum speed is on par with RTX 3070's theoretical maximum speed.
Nothing runs at their theoretical maximum speeds though.

**Comment by xam0482** [comment]
我觉得 张量提升速度和 兼容 ，AUTOMATIC1111和nv已经在做了，等他们实现了再做也不晚，都去做的话浪费人力，而且只能在30系40系的nv卡上应用。
新模型应用倒是能提生新鲜感体验 可以普惠万众，毕竟有没有，和有多少的区别还是挺大的

**Reply by ghost** [reply]
Microsoft Olive适用于20xx系列的显卡，我在这里找到了这个评论：

https://github.com/microsoft/Olive/issues/306#issuecomment-1565643628

我不确定这是否也适用于TensorRT。但我认为应该是的。是的，如果两个开发者现在都在寻找解决方案，然后Nvidia发布了一些内容，那将是很糟糕的情况。但我几乎等不及将TensorRT整合到我的工作流程中了。

**Reply by vladmandic** [reply]
guys, i'd love to know what you're talking about - especially since this is the thread i've started? if you're commenting on a thread that is written in english, comment in english. otherwise, feel free to start a new thread in any language of your choice.

**Comment by sidonsoft** [comment]
TensorRT or riot <3

**Reply by iDeNoh** [reply]
Bring on the riots, daddy wants a new set of models!

**Comment by sevenlucas7** [comment]
For me, as a macOS M1 user, I can only vote for the second option. Haha, it's partly because it brings a new experience and creativity with the new models, and partly because option two utilizes all GPUs. I also have a request: there is really limited optimization and research about running SD on Mac available on the internet, and I don't possess that technical knowledge. I can only hope for technical experts like you, Vlado, to provide insights and guidance.

**Reply by sidonsoft** [reply]
vlad can do both, he is just being lazy :P

**Reply by sevenlucas7** [reply]
Hahaha, I can only say Vlado, keep it up!!! I need you!!! Especially for those of us who have only 8GB on our Macs, we deeply regret not getting the 16GB option...

**Reply by sidonsoft** [reply]
I made sure as hell to get 16gb unified for my m2

**Reply by sevenlucas7** [reply]
Oh, I deeply regret getting only 8GB for my M2.

**Reply by sevenlucas7** [reply]
When I enable four  LORA, adetailer, controlnet, and tile - in SD, it runs at an "astonishing speed."
<img width="590" alt="image" src="https://github.com/vladmandic/automatic/assets/131904649/207e7431-9d8d-4743-956b-9ed7a026e8ef">
